File: 第2章 几何绘图/2.6 十字形图案/画十字形图案.cpp
/*
程序：画十字形图案 
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
// 画基本图形十字形
void draw_figure()
{
	for (int i = 0; i < 4; i++) {
		pen.fd(50);
		pen.rt(90);
		pen.fd(50);
		pen.lt(90);
		pen.fd(50);
		pen.rt(90);
	}
}

// 画十字形图案
int main()
{
	pen.color(_black);
	pen.size(4);
	pen.angle(0);
    int t = 4;                    // 个数
	for (int i = 0; i < t; i++) { // 旋转画图
		draw_figure();
		pen.lt(360.0/t);
	}
	return 0;
}

File: 第2章 几何绘图/2.6 十字形图案/画十字形图案.py
'''
程序：画十字形图案
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
from turtle import *

def draw_figure():
    '''画基本图形十字形'''
    for i in range(4):
        fd(50)
        right(90)
        fd(50)
        left(90)
        fd(50)
        right(90)

def main():
    '''画十字形图案'''
    mode('logo')
    pencolor('black')
    pensize(4)
    seth(0)
    t = 4              # 个数
    for i in range(t): # 旋转画图
        draw_figure()
        left(360/t)

if __name__ == '__main__':
    main()
